packages/react-native-sdk/src/components/call/CallTopView/CallTopView.tsx

```diff
@@ -0,0 +1,132 @@
+} from 'react-native';
+import { ParticipantsInfoBadge } from './ParticipantsInfoBadge';
+import { theme } from '../../../theme';
+import { Back } from '../../../icons/Back';
+import { Z_INDEX } from '../../../constants';
+import { TopViewBackground } from '../../../icons';
+import { useCallStateHooks, useI18n } from '@stream-io/video-react-bindings';
+import { CallingState } from '@stream-io/video-client';
+
+export type CallTopViewProps = {
+  /**
+   * Handler to be called when the back button is pressed in the CallTopView.
+   * @returns void
+   */
+  onBackPressed?: () => void;
+  /**
+   * Handler to be called when the Participant icon is pressed in the CallTopView.
+   * @returns
+   */
+  onParticipantInfoPress?: () => void;
+  /**
+   * Title to be rendered at the center of the Header.
+   */
+  title?: string;
+  /**
+   * Style to override the container of the CallTopView.
+   */
+  style?: StyleProp<ViewStyle>;
+};
+
+export const CallTopView = ({
+  onBackPressed,
+  onParticipantInfoPress,
+  title,
+  style,
+}: CallTopViewProps) => {
+  const { useCallCallingState } = useCallStateHooks();
+  const callingState = useCallCallingState();
+  const { t } = useI18n();
+  const [headerHeight, setHeaderHeight] = useState<number>(0);
+  const isCallReconnecting = callingState === CallingState.RECONNECTING;
+
+  const onLayout: React.ComponentProps<typeof View>['onLayout'] = (event) => {
+    const { height } = event.nativeEvent.layout;
+    setHeaderHeight(height);
+  };
+
+  return (
+    <View style={[styles.container, style]}>
+      {/* Component for the background of the CallTopView. Since it has a Linear Gradient, an SVG is used to render it. */}
+      <TopViewBackground height={headerHeight} width={'100%'} />
+      <View style={styles.topView} onLayout={onLayout}>
+        <View style={styles.leftElement}>
+          {onBackPressed && (
+            <Pressable
+              style={({ pressed }) => [
+                theme.icon.md,
+                styles.backIcon,
+                { opacity: pressed ? 0.2 : 1 },
+              ]}
+              onPress={onBackPressed}
+              disabled={!onBackPressed}
+            >
+              <Back color={theme.light.static_white} />
+            </Pressable>
+          )}
+        </View>
+        <View style={styles.centerElement}>
+          {title ? (
+            <Text style={styles.title} numberOfLines={1}>
+              {title}
+            </Text>
+          ) : (
+            isCallReconnecting && (
+              <Text style={styles.title}>{t('Reconnecting...')}</Text>
+            )
+          )}
+        </View>
+        <View style={styles.rightElement}>
+          {onParticipantInfoPress && (
+            <ParticipantsInfoBadge
+              onParticipantInfoPress={onParticipantInfoPress}
+            />
+          )}
+        </View>
+      </View>
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    ...StyleSheet.absoluteFillObject,
```

Comment:
why is it filled fully here, i don't get it

Reply:
it should only have 

```
{ 
  position: 'absolute',
  top: 0,
  left: 0
}
```

Reply:
does the above break it?

Reply:
I don't want to explicitly use `width: "100%"`  but I see this isn't ideal. Removing it.

Reply:
> does the above break it?

Yes, the width won't be 100%

Reply:
{ position: 'absolute', top: 0, left: 0, right: 0 } maybe?

Reply:
Fixed ✅